sql/src/main/java/io/crate/execution/dml/upsert/InsertSourceGen.java

```diff
@@ -51,4 +55,8 @@ static InsertSourceGen of(TransactionContext txnCtx,
         }
         return new InsertSourceFromCells(txnCtx, functions, table, indexName, validation, targets);
     }
+
+    default BytesReference toBytesReference(Map<String, Object> source) throws IOException {
```

Comment:
Would it be possible to add the default implementation to `generateSourceAndCheckConstraintsAsBytesReference` instead of adding an additional method?